### PR TITLE
Horizontal

### PR DESCRIPTION
**Pull request by sfranky**
matrices can now be printed horizontally instead of vertically.
+ option to override QTOPCONF_YAML key-value through cmdline switch

### CODE REVIEW

File: common_module.py [hunk 1]
```diff
@@ -229,9 +229,12 @@ def process_line(re_search, line, re_match_positions):
 parser.add_option("-F", "--ForceNames", action="store_true", dest="FORCE_NAMES", default=False,
                   help="force names to show up instead of numbered WNs even for very small numbers of WNs")
 # parser.add_option("-f", "--setCOLORMAPFILE", action="store", type="string", dest="COLORFILE")
+parser.add_option("-f", "--setCUSTOMCONFFILE", action="store", type="string", dest="CONFFILE")
 parser.add_option("-m", "--noMasking", action="store_true", dest="NOMASKING", default=False,
                   help="Don't mask early empty WNs (default: if the first 30 WNs are unused, counting starts from 31).")
-parser.add_option("-o", "--SetVerticalSeparatorXX", action="store", dest="WN_COLON", default=0,
+parser.add_option("-o", "--option", action="store", dest="OPTION", type="string", default=None,
+                  help="Override respective option in QTOPCONF_YAML file")
+parser.add_option("-V", "--SetVerticalSeparatorXX", action="store", dest="WN_COLON", default=0,
```
Comment (fgeorgatos) [inline common_module.py]: why not turn that into a tunable option, just like `transpose_wn_matrices`?

fi. `--option SetVerticalSeparator='|'`

File: qtop.py [hunk 2]
```diff
@@ -92,7 +92,7 @@ def decide_remapping(cluster_dict, _all_letters, _all_str_digits_with_empties):
         exotic_starting = min(cluster_dict['workernode_list']) >= config['exotic_starting_wn_nr'] and \
             'the first starting numbering of a WN is very high and thus would require too much unused space' or False
         percentage_unassigned = len(cluster_dict['_all_str_digits_with_empties']) != len(cluster_dict['all_str_digits']) and \
-            'more than %s*nodes have no jobs assigned' %options.PERCENTAGE or False
+            'more than %s*nodes have no jobs assigned' % float(config['percentage']) or False
```
Comment (fgeorgatos) [inline qtop.py]: jobs?!

File: qtop.py [hunk 3]
```diff
@@ -203,25 +203,34 @@ def display_job_accounting_summary(cluster_dict, total_running_jobs, total_queue
...
-
-    print 'Queues: | ',
+    print colorize('===> ', '#') + colorize('Job accounting summary', '', 'Yellow') + colorize(' <=== ', '#') + colorize(
+        '(Rev: 3000 $) %s WORKDIR = %s' % (colorize(str(datetime.datetime.today()), '', 'Purple'), QTOPPATH),
+        'account_not_colored')
+
+    print '%(Usage Totals)s:\t%(online_nodes)s/%(total_nodes)s %(Nodes)s | %(working_cores)s/%(total_cores)s %(Cores)s |' \
+          '   %(total_run_jobs)s+%(total_q_jobs)s %(jobs)s (R + Q) reported by qstat -q' % \
+          {
+              'Usage Totals': colorize('Usage Totals', '', 'Yellow'),
```
Comment (fgeorgatos) [inline qtop.py]: order parameters as 1,3,2, to economize on code

File: qtop.py [hunk 4]
```diff
@@ -465,20 +486,24 @@ def calculate_wnid_lines(start, stop, highest_wn, wn_vert_labels, **kwargs):
 
         # for longer full-labeled wn ids, add more end-labels (far-right) towards the bottom
         for num in range(8, len(wn_vert_labels) + 1):
-            end_labels.setdefault(str(num), end_labels['7'] + num * ['={___ID___}'])
+            _end_labels.setdefault(str(num), _end_labels['7'] + num * ['={___ID___}'])
```
Comment (fgeorgatos) [inline qtop.py]: _var ... why?

File: qtop.py [hunk 5]
```diff
@@ -529,50 +548,48 @@ def display_remaining_matrices(
...
-        print_char_stop,
-        wn_vert_labels,
-        core_user_map,
-        pattern_of_id,
-        workernodes_occupancy):
+def display_matrix(workernodes_occupancy):
     """
     occupancy_parts needs to be redefined for each matrix, because of changed parameter values
     """
+    global transposed_matrices
```
Comment (fgeorgatos) [inline qtop.py]: why global?

File: qtop.py [hunk 6]
```diff
@@ -964,6 +1016,8 @@ def filter_list_out(batch_nodes, _list=None):
 def filter_list_out_by_name(batch_nodes, _list=None):
     if not _list:
         _list = []
+    else:
+        _list[:] = [eval(i) for i in _list]
```
Comment (fgeorgatos) [inline qtop.py]: _var ?

File: qtop.py [hunk 7]
```diff
@@ -997,6 +1065,25 @@ def filter_list_out_by_name_pattern(batch_nodes, _list=None):
     return batch_nodes
 
 
+def filter_list_in_by_name_pattern(batch_nodes, _list=None):
+    if not _list:
+        _list = []
+    # else:
+    #     _list[:] = [eval(i) for i in _list]
```
Comment (fgeorgatos) [inline qtop.py]: drop comments

File: qtopconf.yaml [hunk 8]
```diff
@@ -138,6 +138,7 @@ sorting:
 
 filtering:
 #  - list_out: [0, 1, 2, 3, 16]
+#   - list_in_by_name: ['gaia-73', 'gaia-80', 'gaia-81']
```
Comment (fgeorgatos) [inline qtopconf.yaml]: drop

File: qtopconf.yaml [hunk 9]
```diff
@@ -219,4 +220,15 @@ extract_info:
...
+# The latter is useful when there are a lot of users starting with the same letter.
 fill_with_user_firstletter: False
+# If set to true, lines represent worker nodes, wherease if set to false, columns do.
+transpose_wn_matrices: False
+occupancy_column_order:
+  - wnid_lines
+  - core_map
+  - Node state
+  - queue name
+horizontal_separator: '|'
```
Comment (fgeorgatos) [inline qtopconf.yaml]: hit enter!!!

File: qtop.py [hunk 10]
```diff
@@ -2,7 +2,7 @@
 
 ################################################
 #              qtop v.0.8.1                    #
```
Comment (fgeorgatos) [inline qtop.py]: advance minor?

File: common_module.py (hunk 1 above)
Comment (sfranky) [inline common_module.py]: this was actually an option that inserts a separator every XX columns. But it was put in the conf file anyway.

File: qtop.py (hunk 3 above)
Comment (sfranky) [inline qtop.py]: fixed

File: qtop.py (hunk 4 above)
Comment (sfranky) [inline qtop.py]: renamed var to something more meaningful

File: qtop.py (hunk 5 above)
Comment (sfranky) [inline qtop.py]: errr maybe that wasn't needed after all...just had to remove it !!

File: qtop.py (hunk 6 above)
Comment (sfranky) [inline qtop.py]: umph...fixed

File: qtop.py (hunk 7 above)
Comment (sfranky) [inline qtop.py]: ok

File: qtopconf.yaml (hunk 8 above)
Comment (sfranky) [inline qtopconf.yaml]: need to leave there for experiments

File: qtopconf.yaml (hunk 9 above)
Comment (sfranky) [inline qtopconf.yaml]: adjusted my editor to take care of this :)

File: qtop.py (hunk 10 above)
Comment (sfranky) [inline qtop.py]: ok

File: qtop.py (hunk 2 above)
Comment (sfranky) [inline qtop.py]: are down or offline I meant!!

**Pull request merged**